purchase_delivery_split_date/models/purchase.py

```diff
@@ -70,6 +70,49 @@ def _create_stock_moves(self, picking):
                 picking)
         return moves
 
+    @api.multi
+    def write(self, values):
+        res = super().write(values)
+        if 'date_planned' in values:
+            self.mapped('order_id')._check_split_pickings()
+        return res
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    def _check_split_pickings(self):
+        for order in self:
+            moves = self.env['stock.move'].search([
+                ('purchase_line_id', 'in', self.order_line.ids),
+                ('state', 'not in', ('cancel', 'done')),
+            ])
+            pickings = moves.mapped('picking_id')
+            pickings_by_date = {}
+            for pick in pickings:
+                pickings_by_date[pick.scheduled_date.date()] = pick
```

Comment:
Could you get multiple pickings with the same scheduled_date here ?

Reply:
we could with manual date changes. In that case the moves would be grouped in one of them. I think.

Reply:
So yes, no risk to accidentally skip some pickings here.